consensus/signatures.go

```diff
@@ -123,44 +119,77 @@ func (s *State) validSignatures(t Transaction) (err error) {
 	}
 
 	// Create the InputSignatures object for each input.
-	sigMap := make(map[OutputID]*InputSignatures)
+	sigMap := make(map[string]*InputSignatures)
```

Comment:
3 different types of objects now have UnlockConditions, and each of these objects points to a parent with a different type of id. The signature package avoids this messyness by casting the ids to a string, resulting in a more generic validation of signatures.